File: 자프링-GraphQL-실습/src/main/java/com/junyss/graphqltest/api/role/model/entity/Role.java
package com.junyss.graphqltest.api.role.model.entity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Entity
public class Role {

	@Id @GeneratedValue
	private String roleId;
	private String job;
	private String requirement;

	@Builder
	public static Role toEntity(String roleId, String job, String requirement) {
		return new Role(roleId, job, requirement);
	}

	public void updateJob(String job) {
		this.job = job;
	}

	public void updateRequirement(String requirement) {
		this.requirement = requirement;
	}
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/role/model/entity/Role.kt
package com.junyharangstudy.kotlingraphqltest.api.role.model.entity

import jakarta.persistence.Entity
import jakarta.persistence.Id

@Entity
class Role (
    @Id var roleId: String,
    var job: String,
    var requirement: String
) {
    fun updateJob(job: String) {
        this.job = job
    }

    fun updateRequirment(requirement: String) {
        this.requirement = requirement
    }
}